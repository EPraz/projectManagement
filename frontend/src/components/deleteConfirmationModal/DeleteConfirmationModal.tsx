import React from "react";
import {
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Button,
  Typography,
} from "@mui/material";
import { DeleteConfirmationModalProps } from "../../types";

const DeleteConfirmationModal: React.FC<DeleteConfirmationModalProps> = ({
  open,
  onClose,
  onConfirm,
  itemName,
  disabled = false,
}) => {
  const handleSubmit = () => {
    onConfirm();
    onClose();
  };
  return (
    <Dialog open={open} onClose={onClose} maxWidth="xs" fullWidth>
      <DialogTitle>Confirm Delete</DialogTitle>
      <DialogContent>
        <Typography>
          Are you sure you want to delete <strong>{itemName}</strong>?
          <br />
          <br />
          <em style={{ fontSize: "smaller" }}>This action cannot be undone.</em>
        </Typography>
      </DialogContent>
      <DialogActions>
        <Button
          onClick={onClose}
          color="primary"
          variant="outlined"
          disabled={disabled}
        >
          Cancel
        </Button>
        <Button
          onClick={handleSubmit}
          //   color="primary"
          variant="contained"
          sx={{
            backgroundColor: "green",
            color: "white",
            ":hover": { backgroundColor: "white", color: "green" },
          }}
          disabled={disabled}
        >
          Proceed
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default DeleteConfirmationModal;
