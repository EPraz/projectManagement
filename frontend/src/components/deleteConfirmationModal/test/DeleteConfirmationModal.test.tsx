// DeleteConfirmationModal.test.tsx
import { render, screen, fireEvent } from "@testing-library/react";
import DeleteConfirmationModal from "../DeleteConfirmationModal";
import "@testing-library/jest-dom";

describe("DeleteConfirmationModal", () => {
  const mockOnClose = jest.fn();
  const mockOnConfirm = jest.fn();
  const itemName = "Test Item";

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("match snapshot", () => {
    const { asFragment } = render(
      <DeleteConfirmationModal
        open={true}
        onClose={mockOnClose}
        onConfirm={mockOnConfirm}
        itemName={itemName}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the dialog with correct content", () => {
    render(
      <DeleteConfirmationModal
        open={true}
        onClose={mockOnClose}
        onConfirm={mockOnConfirm}
        itemName={itemName}
      />
    );

    expect(screen.getByText("Confirm Delete")).toBeInTheDocument();
    expect(screen.getByText(new RegExp(itemName, "i"))).toBeInTheDocument();
    expect(
      screen.getByText(/This action cannot be undone/i)
    ).toBeInTheDocument();
  });

  it("calls onClose when Cancel is clicked", () => {
    render(
      <DeleteConfirmationModal
        open={true}
        onClose={mockOnClose}
        onConfirm={mockOnConfirm}
        itemName={itemName}
      />
    );
    fireEvent.click(screen.getByText(/cancel/i));
    expect(mockOnClose).toHaveBeenCalled();
  });

  it("calls onConfirm and then onClose when Proceed is clicked", () => {
    render(
      <DeleteConfirmationModal
        open={true}
        onClose={mockOnClose}
        onConfirm={mockOnConfirm}
        itemName={itemName}
      />
    );
    fireEvent.click(screen.getByText(/proceed/i));
    expect(mockOnConfirm).toHaveBeenCalled();
    expect(mockOnClose).toHaveBeenCalled();
  });
});
