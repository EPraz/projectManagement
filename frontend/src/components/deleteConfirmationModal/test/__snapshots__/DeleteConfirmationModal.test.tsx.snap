// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirmationModal match snapshot 1`] = `<DocumentFragment />`;
